### PR TITLE
[CALCITE-4999] ARRAY, MULTISET functions should return a collection of scalars if a sub-query returns 1 column

### PR DESCRIPTION
**Pull request by dssysolyatin**

### CODE REVIEW

**Comment by rubenada** [comment]
Could you please provide a unit test with a SQL query? Probably the best option would be adding the problematic query into `big-query.iq`

**Comment by rubenada** [comment]
Kindly reminder: normally the commit message should be the same as the Jira title (in case of a bug, it should normally describe the bug) https://calcite.apache.org/develop/#contributing

**Comment by julianhyde** [comment]
This proposed change is not an "improvement".

File: core/src/main/java/org/apache/calcite/rel/core/Collect.java [hunk 1]
```diff
@@ -132,6 +133,47 @@ public static Collect create(RelNode input,
...
   }
 
+  /**
+   * Creates a Collect.
+   *
+   * @param input          Input relational expression
+   * @param sqlKind        SqlKind
+   * @param fieldName      Name of the sole output field
+   */
+  public static Collect create(RelNode input,
```
Comment (rubenada) [inline core/src/main/java/org/apache/calcite/rel/core/Collect.java:143]: Should we deprecate any of the previous `create` methods to encourage people to use this new one? Or are they all still valid?

Comment (dssysolyatin) [reply core/src/main/java/org/apache/calcite/rel/core/Collect.java:143]: I think yes, `create(RelNode input,SqlTypeName collectionType, String fieldName)` should be deprecated.

Comment (rubenada) [reply core/src/main/java/org/apache/calcite/rel/core/Collect.java:144]: Ok, in that case we should not be calling it anymore.
If I am not mistaken, the PR has removed all the previously existing calls (in favor of the new `create` method), but said new method calls the deprecated one (Collect.java:174): it should call the constructor directly in that line.

Comment (dssysolyatin) [reply core/src/main/java/org/apache/calcite/rel/core/Collect.java:144]: Fixed

**Pull request merged**